# Certificate configuration scenarios.md

```diff
@@ -0,0 +1,246 @@
+# Certificate configuration scenarios
+
+There are five scenarios/use cases for configuring a certificates.
+Use the applicable certificate configuration scenario according to your needs.
+
+* [Scenario 1: Use a PKCS12 (keystore) with Zowe generated certificates](#scenario-1-use-a-pkcs12-keystore-with-zowe-generated-certificates)
+* [Scenario 2: Use a PKCS12 (keystore) and import a certificate generated by another CA](#scenario-2-use-a-pkcs12-keystore-and-import-a-certificate-generated-by-another-ca)
+* [Scenario 3: Use a z/OS Keyring with Zowe generated certificates](#scenario-3-use-a-zos-keyring-with-zowe-generated-certificates)
+* [Scenario 4: Use a z/OS Keyring and connect to an existing certificate](#scenario-4-use-a-zos-keyring-and-connect-to-an-existing-certificate)
+* [Scenario 5: Use a z/OS Keyring and import a certificate stored in a data set](#scenario-5-use-a-zos-keyring-and-import-a-certificate-stored-in-a-data-set)
+
+## Scenario 1: Use a PKCS12 (keystore) with Zowe generated certificates
+
+1. Select the type of certificate storage with `type: PKCS12`  
+Possible values include the following values:  
+
+    * PKCS12  
+    * JCEKS  
+    * JCECCAKS  
+    * JCERACFKS  
+    * JCECCARACFKS  
+    * JCEHYBRIDRACFKS
+
+2. Customize the keystore directory in the following format:
+    ```
+    /var/zowe/keystore
+    ```
+3. Lock the keystore directory so it is accessible only to the Zowe runtime user and group:
+    ```
+    lock: true
+    ```
+4. Customize the certificate alias name. The default value is `localhost`.  
+    **Note:** Use all lower case as an alias.
+5. Set keystore password. Customizing this value is highly recommended. The default value is `password`.
+6. Set the alias name of self-signed certificate authority. The default value is `local_ca`.  
+   **Note:** Use all lower case as an alias.
+   ```
+   caAlias: local_ca
+   ```
+7. Set the password of the keystore stored self-signed certificate authority. The default value is `local_ca_password`.
+    ```
+    caPassword: local_ca_password
+    ```
+8. (Optional) Specify the distinguished name for Zowe generated certificates. 
+
+    ```
+    dname:
+      caCommonName: ""
+      commonName: ""
+      orgUnit: ""
+      org: ""
+      locality: ""
+      state: ""
+      country: ""
+    ```
+9. Set the validity in days for the Zowe generated certificates
+    ```  
+    validity: 3650
+    ```
+10. Set the domain names and IPs specified in nested subsection `SAN`. If this field is not defined, the `zwe init` command uses the value `zowe.externalDomains`.
+    ```
+    san:
+        sample domain name
```

Comment:
`sample domain name` and `sample IP address` are commented in yaml. It would be better to put them after `#` or just remove them.